token/cache.go

```diff
@@ -0,0 +1,93 @@
+package token
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	maxLockWait = 30 * time.Second
```

Comment:
And this is the maximum duration we'll spend holding the lock, not the maximum duration we'll wait for the lock if another goroutine holds it?

Reply:
Yes, probably lockHoldTimeout a better name ?